rasa/core/channels/botframework.py

```diff
@@ -188,6 +201,40 @@ def __init__(self, app_id: Text, app_password: Text) -> None:
         self.app_id = app_id
         self.app_password = app_password
 
+        self.jwt_keys: Dict[Text, Any] = {}
+        self.jwt_update_time = datetime.datetime.fromtimestamp(0)
+
+        self._update_cached_jwk_keys()
+
+    def _update_cached_jwk_keys(self) -> None:
+        logger.debug("Updating JWT keys for the Botframework.")
+        req = requests.get(MICROSOFT_OPEN_ID_URI)
```

Comment:
Nitpicking, I would actually label this variable `response`, since this would hold a Response object.